lib/internal/url.js

```diff
@@ -591,15 +595,19 @@ function getSearchParamPairs(target) {
   return values;
 }
 
+function initSearchParams(url, init) {
+  init = String(init);
+  if (init[0] === '?') init = init.slice(1);
+  url[searchParams] = querystring.parse(init);
+}
+
 class URLSearchParams {
   constructor(init = '') {
     if (init instanceof URLSearchParams) {
       const childParams = init[searchParams];
       this[searchParams] = Object.assign(Object.create(null), childParams);
     } else {
-      init = String(init);
-      if (init[0] === '?') init = init.slice(1);
-      this[searchParams] = querystring.parse(init);
+      initSearchParams(this, init);
```

Comment:
Yeah, I'd much prefer keeping the `String()` and `.slice` here, as they are such in the spec.